zerver/management/commands/runtusd.py

```diff
@@ -0,0 +1,63 @@
+import os
+from typing import Any
+from urllib.parse import SplitResult
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from typing_extensions import override
+
+
+class Command(BaseCommand):
+    help = """Starts the tusd server"""
+
+    @override
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "listen", help="[Port, or address:port, to bind HTTP server to]", type=str
+        )
+        parser.add_argument(
+            "hooks_http",
+            help="[An HTTP endpoint to which hook events will be sent to]",
+            default="http://127.0.0.1/api/internal/tusd",
+            nargs="?",
+        )
+
+    @override
+    def handle(self, *args: Any, **options: Any) -> None:
+        listen = options["listen"]
+        if listen.isdigit():
+            addr, port = "127.0.0.1", int(listen)
+        else:
+            r = SplitResult("", listen, "", "", "")
+            if r.port is None:
+                raise CommandError(f"{listen!r} does not have a valid port number.")
+            addr, port = r.hostname or "127.0.0.1", r.port
+
+        hooks_http = options["hooks_http"]
+
+        tusd_args = [
+            "tusd",
+            "-base-path=/api/v1/tus/",
+            f"-port={port}",
+            f"-host={addr}",
+            "-behind-proxy",
+            f"-hooks-http={hooks_http}",
+            "-hooks-http-forward-headers=Cookie,Authorization",
+            "--hooks-enabled-events=pre-create,pre-finish",
+            "-disable-download",
+        ]
+        env_vars = os.environ.copy()
```

Comment:
I assume we're not passing `-max-size` because we want the `pre-create` hook to enforce that, as what the maximum size we allow may depend on the context/type of file.

Reply:
Correct.  I'll add a comment.